types_test.go

```diff
@@ -1281,3 +1295,232 @@ func TestDecodeDeeplyNestedTuple(t *testing.T) {
+			A int64 `edgedb:"a"`
+		}
+
+		var result NamedTuple
+		err := conn.QueryOne(ctx, "SELECT (a := 1,)", &result)
+		require.Nil(t, err)
+		assert.Equal(t, NamedTuple{A: 1}, result)
+	}
+
+	// decoding into map using reflect
+	{
+		var result []interface{}
+		err := conn.QueryOne(ctx, "SELECT ((a := 1),)", &result)
+		require.Nil(t, err)
+		expected := []interface{}{
+			map[string]interface{}{"a": int64(1)},
+		}
+		assert.Equal(t, expected, result)
+	}
+
+	// decoding into struct using reflect
+	{
+		type NamedTuple struct {
+			A []interface{} `edgedb:"a"`
+		}
+
+		var result NamedTuple
+		err := conn.QueryOne(ctx, "SELECT (a := ((1,),))", &result)
+		require.Nil(t, err)
+
+		expected := NamedTuple{A: []interface{}{[]interface{}{int64(1)}}}
+		assert.Equal(t, expected, result)
+	}
+}
+
+func TestReceveTuple(t *testing.T) {
+	ctx := context.Background()
+
+	// decoding uses pointers when all tuple elements are scalars
+	var result [][]interface{}
+	err := conn.Query(ctx, `SELECT ()`, &result)
+	require.Nil(t, err, "unexpected error: %v", err)
+	require.Equal(t, [][]interface{}{{}}, result)
+
+	err = conn.Query(ctx, `SELECT (<int64>$0,)`, &result, int64(1))
+	require.Nil(t, err, "unexpected error: %v", err)
+	require.Equal(t, [][]interface{}{{int64(1)}}, result)
+
+	err = conn.Query(ctx, `SELECT (1, "abc")`, &result)
+	require.Nil(t, err, "unexpected error: %v", err)
+	require.Equal(t, [][]interface{}{{int64(1), "abc"}}, result)
+
+	err = conn.Query(ctx, `SELECT {(1, "abc"), (2, "def")}`, &result)
+	require.Nil(t, err, "unexpected error: %v", err)
+	expected := [][]interface{}{
+		{int64(1), "abc"},
+		{int64(2), "def"},
+	}
+	require.Equal(t, expected, result)
+
+	// decoding uses reflect when any tuple element is not a scalar
+	err = conn.Query(ctx, `SELECT (1, ("abc",))`, &result)
+	require.Nil(t, err, "unexpected error: %v", err)
+	expected = [][]interface{}{{int64(1), []interface{}{"abc"}}}
+	require.Equal(t, expected, result)
+}
+
+func TestSendAndReceveArray(t *testing.T) {
+	ctx := context.Background()
+
+	var result []int64
+	err := conn.QueryOne(ctx, "SELECT <array<int64>>$0", &result, "hello")
+	assert.EqualError(t, err,
+		"edgedb.InvalidArgumentError: "+
+			"expected args[0] to be []int64 got: string")
+
+	// decoding uses reflect when parent is a tuple
+	var nested []interface{}
+	err = conn.QueryOne(ctx, "SELECT (<array<int64>>$0,)", &nested, []int64{1})
+	require.Nil(t, err)
+	assert.Equal(t, []interface{}{[]int64{1}}, nested)
+
+	// decoding using pointers
+	err = conn.QueryOne(ctx, "SELECT <array<int64>>$0", &result, []int64(nil))
+	require.Nil(t, err)
+	assert.Equal(t, []int64(nil), result)
+
+	err = conn.QueryOne(ctx, "SELECT <array<int64>>$0", &result, []int64{1})
+	require.Nil(t, err)
+	assert.Equal(t, []int64{1}, result)
+
+	arg := []int64{1, 2, 3}
+	err = conn.QueryOne(ctx, "SELECT <array<int64>>$0", &result, arg)
+	require.Nil(t, err)
+	assert.Equal(t, []int64{1, 2, 3}, result)
+}
+
+func TestReceveSet(t *testing.T) {
```

Comment:
typo: "Receve" -> "Receive"  -- occurs in multiple places, I suggest global search/replace.

Reply:
changed